README.md

```diff
@@ -12,6 +12,20 @@ metadata based on `git describe`.
     cd ./docker
     make
 
+## Release instructions
+
+When releasing/tagging a new version please use annotated tags in the
+command line interface.
+
+```bash
+git checkout master
+git tag -a 'x.y.z' -m 'release x.y.z'
+git push origin x.y.z
```

Comment:
Probably also best to `git checkout devel` again, to leave the checkout on a branch other than one that can do release automation

Reply:
See your point. I was assuming all of this was done in a fresh clone but didn't mention it so my bad. A fast-forward merge devel -> master maybe(?) to complain if there are extra things in your local copy of master?

I think I agree with everything there but the last one. If you are working in your own clone, not a fresh one, `push --tags origin` will push all tags you may have locally. Including alphas and betas or whatever else you may have.

Reply:
If you're working in a clone which has direct push access to the main repo, I'd say it's best not to be making any changes there that you don't want to push. Stuff like making extra tags, branches, changes should be done on a fork.

Reply:
> If you're working in a clone which has direct push access to the main repo...

Don't dispute that. But people can and there is no way to stop them. Should we go for `Create a fresh clone of the repo`.

So it can be

```bash
git clone --branch master git@github.com:wtsi-npg/samtools_container.git samtools_container && cd $_
git tag -a 'x.y.z' -m 'release x.y.z'
git push origin x.y.z
```

Reply:
I think that's fine as an example because you _do_ have to clone at least once.

Reply:
Added an extra commit with the change. Would like to rebase and squash before merging, but don't want to drag the thing.

Reply:
Would it be ok if I rebase and squash?